lizard_languages/clike.py

```diff
@@ -135,7 +135,7 @@ def _state_global(self, token):
         if token == "template":
             self._state = self._template_declaration
 
-        elif token in ("struct", "class", "namespace"):
+        elif self.last_token != "." and token in ("struct", "class", "namespace"):
             self._state = self._read_namespace
```

Comment:
Sorry for including that one.
I had reverted back that commit.
It is different issue I will be pushing later, related to Java Class Literals (".class") causing out of memory issue with deep nesting.